java/fury-core/src/main/java/org/apache/fury/serializer/BufferSerializers.java

```diff
@@ -44,6 +44,13 @@ public void write(MemoryBuffer buffer, ByteBuffer value) {
       fury.writeBufferObject(buffer, new BufferObject.ByteBufferBufferObject(value));
     }
 
+    @Override
+    public ByteBuffer copy(ByteBuffer value) {
+      ByteBuffer dst = ByteBuffer.allocate(value.capacity());
+      dst.put(value);
```

Comment:
```suggestion
      dst.put(value.duplicate());
```
We need to `duplicate`, otherwise it will update offset in `ByteBuffer`